src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -115,14 +115,23 @@ public FormValidation doCheckApiUri(@QueryParameter String apiUri) {
                 URL api = new URL(apiUri);
                 GitHub github = GitHub.connectToEnterpriseAnonymously(api.toString());
                 github.checkApiUrlValidity();
-                return FormValidation.ok();
+                LOGGER.log(Level.FINE, "Trying to configure a GitHub Enterprise server");
+                return FormValidation.ok("GitHub Enterprise server verified");
             } catch (MalformedURLException mue) {
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                // For example: https:/api.github.com
+                LOGGER.log(Level.WARNING, "Trying to configure a GitHub Enterprise server: " + apiUri);
```

Comment:
Agreed.  🐛  for the combination of this + JsonParseException both logging the same error message;  each should indicate what is wrong in the log line, and the form validation errors should be different.